compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -182,6 +187,7 @@ impl Context {
         let dfg = &main_func.dfg;
         let entry_block = &dfg[main_func.entry_block()];
         let input_witness = self.convert_ssa_block_params(entry_block.parameters(), dfg)?;
+        self.function_name = Some(main_func.name().to_string());
```

Comment:
I think this would just return "main" everytime? Which is not necessarily more descriptive, it might actually be more confusing.